.github/workflows/connector-performance-command.yml

```diff
@@ -19,15 +19,184 @@ on:
+      - name: Write source-harness credentials
+        run: |
+          ci_credentials connectors-performance/source-harness write-to-storage
+        env:
+          GCP_GSM_CREDENTIALS: ${{ secrets.GCP_GSM_CREDENTIALS }}
+      - name: build harness
+        shell: bash
+        run: |
+          echo "Building... source-harness" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY
+          ./gradlew :airbyte-integrations:connectors-performance:source-harness:build -x check
+      - name: build connector
+        shell: bash
+        run: |
+          echo "Building... ${{github.event.inputs.connector}}" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY # this is a blank line
+          echo "Running ./gradlew :airbyte-integrations:connectors:source-postgres:build -x check" 
+          ./gradlew :airbyte-integrations:connectors:source-postgres:build -x check
+      - name: KIND Kubernetes Cluster Setup
+        uses: helm/kind-action@v1.4.0
+        with:
+          config: ./tools/bin/source-harness-kind-cluster-config.yaml
+      - name: Run harness
+        id: run-harness
+        shell: bash
+        env:
+          CONN: ${{ github.event.inputs.connector }}
+          DS: ${{ github.event.inputs.dataset }}
+        run: |
+          echo "Setting admin account permissions"
+          kubectl apply -f ./tools/bin/admin-service-account.yaml
+          echo "Loading connector image into kind" >> $GITHUB_STEP_SUMMARY
+          echo "" >> $GITHUB_STEP_SUMMARY # this is a blank line
+          kind load docker-image airbyte/source-postgres:dev --name chart-testing
+          echo "Loading harness image into kind" >> $GITHUB_STEP_SUMMARY
+          kind load docker-image airbyte/source-harness:dev --name chart-testing
+          echo "loading pod..." 
+          export CONNECTOR_IMAGE_NAME=${CONN/connectors/airbyte}:dev
+          export DATASET=$DS
+          envsubst < ./tools/bin/source-harness-process.yaml | kubectl create -f -
+          echo "get pods..."
+          kubectl get po
+          POD=$(kubectl get pod -l app=source-harness -o jsonpath="{.items[0].metadata.name}")
+          echo "pod name is $POD"
+          #sleep 3
+          #kubectl get events --sort-by=.metadata.creationTimestamp
+          kubectl wait --for=condition=Ready --timeout=20s "pod/$POD"
+          kubectl logs --follow $POD
+          echo "reading end of log"
+          kubectl logs --tail=5 $POD
+          # RUN_RESULT=$(kubectl logs --tail=5 $POD 2>&1)
+          prefix='{"type":"LOG","log":{"level":"INFO","message":"INFO i.a.i.p.PerformanceTest(runTest):165'
+          suffix='"}}'
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "RUN_RESULT<<$EOF" >> $GITHUB_OUTPUT
+          kubectl logs --tail=1 $POD | while read line ; do line=${line#"$prefix"}; line=${line%"$suffix"}; echo $line >> $GITHUB_OUTPUT ; done 
+          echo "$EOF" >> $GITHUB_OUTPUT
```

Comment:
Thinking out loud: what about turning this into a script? 

1) This way, someone can run the script locally if needed.
2) The GHA workflow files mostly deal with orchestration logic.

Reply:
I agree with everything - it makes more sense to have this as a separate script.
But if pl let me do this in the next stage: I need to hook the harness to some daily process and collect the result (more on that offline)
For now I sanitized the code and the whole thing is ~10 lines